### PR TITLE
Validate attachment response and throw error

### PR DESCRIPTION
**Pull request by kraju3**
# Description
At the moment we are not returning a NylasAPIError exception if we are unable to download the attachment. Which results in customers writing the content to a file. This PR adds a validation for the download request and return an exception if the status code is >=400. 

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: nylas/handler/http_client.py [hunk 1]
```diff
@@ -126,9 +130,9 @@ def _execute_download_request(
 
             # If we stream an iterator for streaming the content, otherwise return the entire byte array
             if stream:
-                return response
+                return _validate_download_response(response,stream)
 
-            return response.content if response.content else None
+            return _validate_download_response(response) if response.content else None
```
Comment (AaronDDM) [inline nylas/handler/http_client.py:135]: @kraju3 do we need to make this change here? Looks like if it's `stream` we handle the response differently earlier in this function anyways.

File: nylas/handler/http_client.py [hunk 2]
```diff
@@ -126,9 +130,9 @@ def _execute_download_request(
 
             # If we stream an iterator for streaming the content, otherwise return the entire byte array
             if stream:
-                return response
+                return _validate_download_response(response,stream)
```
Comment (AaronDDM) [inline nylas/handler/http_client.py:133]: Maybe we can simplify this?

```suggestion
                return response if response.ok else response.content
```

Comment (AaronDDM) [reply nylas/handler/http_client.py:133]: But then, will `_execute_download_request` not error out where ever it is used 🤔 So I guess the question is, is this the correct place to do this validate this?

File: nylas/handler/http_client.py (hunk 1 above)
Comment (kraju3) [reply nylas/handler/http_client.py:135]: Hey @AaronDDM I was hoping to keep the same behavior plus return an error if the error code is different

**Comment by kraju3** [comment]
@AaronDDM Is it good to merge this PR?

**Review by AaronDDM — APPROVED** [review]

**Pull request merged**